src/lib/translate.js

```diff
@@ -2,42 +2,51 @@ import delve from 'dlv';
- *
- *	@private
- *	@param {String} id				Intl field name/id (subject to scope)
- *	@param {String} [scope='']		Scope, which prefixes `id` with `${scope}.`
- *	@param {Object} dictionary		A nested object containing translations
- *	@param {Object} [fields={}]		Template fields for use by translated strings
- *	@param {Number} [plural]		Indicates a quantity, used to trigger pluralization
- *	@param {String|Array} [fallback]	Text to return if no translation is found
- *	@returns {String} translated
- */
-export default function translate(id, scope, dictionary, fields, plural, fallback) {
-	if (scope) id = scope + '.' + id;
-
-	let value = dictionary && delve(dictionary, id);
 
+/**
+ * Default function to determine what plural form to use from the provided dictionary
+ * @param dict the dictorary with possible plural forms
+ * @param plural the plural count
+ * @returns {String | false}
+ */
+export function defaultProvider(dict, plural) {
 	// plural forms:
 	// key: ['plural', 'singular']
 	// key: { none, one, many }
 	// key: { singular, plural }
-	if ((plural || plural===0) && value && typeof value==='object') {
-		if (value.splice) {
-			value = value[plural] || value[0];
-		}
-		else if (plural===0 && defined(value.none)) {
-			value = value.none;
+	if ((plural || plural === 0) && dict && typeof dict === 'object') {
+		if (dict.splice) {
+			return dict[plural] || dict[0];
 		}
-		else if (plural===1 && defined(value.one || value.singular)) {
-			value = value.one || value.singular;
+		else if (plural === 0 && defined(dict.none)) {
+			return dict.none;
 		}
-		else {
-			value = value.some || value.many || value.plural || value.other || value;
+		else if (plural === 1 && defined(dict.one || dict.singular)) {
+			return dict.one || dict.singular;
 		}
+		return dict.some || dict.many || dict.plural || dict.other || dict;
 	}
+}
+
+/** Attempts to look up a translated value from a given dictionary.
+ *  Also supports json templating using the format: {{variable}}
+ *    Falls back to default text.
+ *
+ *    @private
+ *    @param {String} id                Intl field name/id (subject to scope)
+ *    @param {String} [scope='']        Scope, which prefixes `id` with `${scope}.`
+ *    @param {Object} dictionary        A nested object containing translations
+ *    @param {Object} [fields={}]        Template fields for use by translated strings
+ *    @param {Number} [plural]        Indicates a quantity, used to trigger pluralization
+ *    @param {String|Array} [fallback]    Text to return if no translation is found
+ *    @param provider {function: String | false} Provider function to extract the plural form from the dictionary
+ *    @returns {String} translated
+ */
+export default function translate(id, scope, dictionary, fields, plural, fallback, provider = defaultProvider) {
+	if (scope) id = scope + '.' + id;
+
+	let value = dictionary && delve(dictionary, id);
+	value = provider(value, plural) || defaultProvider(value, plural);
```

Comment:
Since we are using the name `Provider` in `IntlProvider` this name might be confusing, maybe we can call it a "pluralizer"?